### PR TITLE
ENH: `dpnp` interop for SPMD `sklearnex` interface

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
* Added `dpnp` interop for scikit-learn-intelex SPMD interfaces
* Examples for `examples/sklearnex/random_forest_regressor_spmd.py` were updated.

### CODE REVIEW

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee2fe86d-2ba9-f163-8d3e-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU.

**Comment by samir-nasibli** [comment]
> Job: http://intel-ci.intel.com/ee2fe86d-2ba9-f163-8d3e-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU.

All related examples validation are passed except `random_forest_classifier_spmd.py` that is ongoing issue and not related with the PR's changes.

```python
##### /examples/sklearnex/random_forest_classifier_dpctl_batch.py
	PASSED	random_forest_classifier_dpctl_batch.py

##### /examples/sklearnex/basic_statistics_spmd.py
	PASSED	basic_statistics_spmd.py

##### /examples/sklearnex/random_forest_regressor_dpnp_batch.py
	PASSED	random_forest_regressor_dpnp_batch.py

##### /examples/sklearnex/pca_spmd.py
	PASSED	pca_spmd.py

##### /examples/sklearnex/random_forest_classifier_spmd.py
	FAILED	random_forest_classifier_spmd.py	with errno	1

##### /examples/sklearnex/random_forest_regressor_spmd.py
	PASSED	random_forest_regressor_spmd.py

##### /examples/sklearnex/linear_regression_spmd.py
	PASSED	linear_regression_spmd.py

##### /examples/sklearnex/knn_bf_classification_dpnp_batch.py
	PASSED	knn_bf_classification_dpnp_batch.py

##### /examples/sklearnex/knn_bf_classification_spmd.py
	PASSED	knn_bf_classification_spmd.py

##### /examples/sklearnex/knn_bf_regression_spmd.py
	PASSED	knn_bf_regression_spmd.py

##### /examples/sklearnex/kmeans_spmd.py
	PASSED	kmeans_spmd.py
107/108 examples passed/skipped, 1 failed
```

Updated in the PR `random_forest_regressor_spmd.py` example also passed validation.

File: examples/sklearnex/random_forest_regressor_spmd.py
Comment (Alexsandruss) [inline examples/sklearnex/random_forest_regressor_spmd.py:1]: Why this spmd example is the only changed? And what is the reason of classifier example failure?

File: onedal/_device_offload.py [hunk 1]
```diff
@@ -61,7 +68,13 @@ def wrapper_impl(obj, *args, **kwargs):
                 hostkwargs["queue"] = data_queue
                 result = _run_on_device(func, obj, *hostargs, **hostkwargs)
                 if usm_iface is not None and hasattr(result, "__array_interface__"):
-                    return _copy_to_usm(data_queue, result)
+                    result = _copy_to_usm(data_queue, result)
+                    if (
+                        dpnp_available
+                        and len(args) > 0
+                        and isinstance(args[0], dpnp.ndarray)
```
Comment (Alexsandruss) [inline onedal/_device_offload.py:75]: Warning is required if `args` types are inconsistent.

**Review by KulikovNikita — APPROVED** [review]

File: examples/sklearnex/random_forest_regressor_spmd.py
Comment (samir-nasibli) [reply examples/sklearnex/random_forest_regressor_spmd.py:1]: > And what is the reason of classifier example failure?

This is related with RF issues, I didn't investigate it yet. The same fail is on the master branch. Current PR is not targeted to fix this particular issue.

**Review by inteldimitrius — COMMENTED** [review]
Changes look good to me, but I don't have sufficient rights to approve

File: examples/sklearnex/random_forest_regressor_spmd.py
Comment (Alexsandruss) [reply examples/sklearnex/random_forest_regressor_spmd.py:1]: I asked about all spmd examples, not other non-spmd. Users might not find these dpnp.ndarray/dpctl.tensor feature if it's mentioned in only one example

Comment (samir-nasibli) [reply examples/sklearnex/random_forest_regressor_spmd.py:1]: I think all related features should be properly described in docs. For SPMD and non-spmd interfaces docs will be updated.

**Pull request merged**